libyul/backends/wasm/BinaryTransform.cpp

```diff
@@ -530,8 +538,8 @@ bytes BinaryTransform::functionSection(vector<FunctionDefinition> const& _functi
 bytes BinaryTransform::memorySection()
 {
 	bytes result = lebEncode(1);
-	result.push_back(0); // flags
-	result.push_back(1); // initial
+	result.push_back(static_cast<uint8_t>(Flags::MaximumFieldOmitted));
```

Comment:
If you want to be precise this is more like an enum of limit kind and not "MaximumFieldOmitted".

If you insist the better version would be:
```
class enum LimitsKind {
  Min = 0x00,
  MinMax = 0x01,
}
```

Reply:
Hmm... yeah, you could look at it that way. I based mine on the spec which treats it as a flag saying whether the `max` field is present:

https://webassembly.github.io/spec/core/binary/types.html#limits
> Limits are encoded with a preceding flag indicating whether a maximum is present.

Either interpretation is fine for me. Should I change this to two separate enums?

Reply:
I split the enum into `LimitsKind` and `Mutability`. See if it's better now.

Actually, I think your version reads better than just a bunch of random flags so it's an improvement.